examples/test-scripts/kubernetes/k8s-helloworld.yaml

```diff
@@ -0,0 +1,117 @@
+# Copyright 2016 IBM Corporation
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+apiVersion: v1
+kind: Service
+metadata:
+  name: gateway
+spec:
+  ports:
+  - port: 6379
+    targetPort: 6379
+    nodePort: 32000
+    protocol: TCP
+  selector:
+    name: gateway
+  type: NodePort
+---
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: gateway
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        name: gateway
+    spec:
+      containers:
+      - name: gateway
+        image: amalgam8/a8-sidecar:latest
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 6379
+        args:
+        - /usr/bin/run_filebeat.sh
+        env:
+        - name: A8_PROXY
+          value: "true"
+        - name: A8_CONTROLLER_URL
+          value: http://$(CONTROLLER_SERVICE_HOST):$(CONTROLLER_SERVICE_PORT)
+        - name: A8_CONTROLLER_POLL
+          value: 5s
+        - name: A8_SERVICE
+          value: gateway:none
+        - name: A8_DISCOVERY_ADAPTER
+          value: kubernetes
+        - name: A8_ELASTICSEARCH_SERVER
+          value: $(LOGSERVER_SERVICE_HOST):9200
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: helloworld
+spec:
+  clusterIP: None
+  ports:
+  - port: 5000
+    protocol: TCP
+    name: http
+  selector:
+    name: helloworld
+---
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: helloworld-v1
+spec:
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        name: helloworld
+        version: v1
+    spec:
+      containers:
+      - name: helloworld
+        image: amalgam8/a8-examples-helloworld-v1:{APP_VER}
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 5000
+        env:
+        - name: A8_SERVICE
+          value: helloworld:version=v1
+---
+apiVersion: v1
+kind: ReplicationController
+metadata:
+  name: helloworld-v2
+spec:
+  replicas: 2
+  template:
+    metadata:
+      labels:
+        name: helloworld
+        version: v2
+    spec:
+      containers:
+      - name: helloworld
+        image: amalgam8/a8-examples-helloworld-v2:{APP_VER}
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 5000
+        env:
+        - name: A8_SERVICE
+          value: helloworld:version=v2
```

Comment:
If this is the only difference in all these files, why can't we use bunch of environment variables ?